### DISCUSSION TITLE
Issues when running `dolphin` with ISCE2 stackStripmap products

### DISCUSSION THREAD

**Discussion by david-ncu2019** [isce-framework/dolphin#482] [Q&A]
My input for ISCE2 `stackStripmap` was the TerraSAR-X images covering the central part of Taiwan. I applied stackStripmap `slc workflow`, aiming to use the output for running `miaplpy` as well. There were 25 scenes in total.

I applied the `dolphin config --slc-files ../merged/SLC/*/*.slc`, following the instructions from dolphin Github page.
(full command: `dolphin config --slc-files ../ISCE_output/merged/SLC/*/*.slc --strides 4 10 --output-bounds 120.287626 23.558988 120.315822 23.591824 --work-directory testrun001`)

--------

After running `dolphin run dolphin_config.yaml`, I ended up with the following errors (the detailed log is attached at the bottom):

```
    File "/home/davidncu/miniconda3/envs/dolphin-env/lib/python3.12/site-packages/dolphin/io/_core.py", line 304, in get_raster_crs
        print(CRS.from_wkt(ds.GetProjection()))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "/home/davidncu/miniconda3/envs/dolphin-env/lib/python3.12/site-packages/pyproj/crs/crs.py", line 433, in from_wkt
        raise CRSError(f"Invalid WKT string: {in_wkt_string}")
pyproj.exceptions.CRSError: Invalid WKT string: 
```
I tried to check the source of the errors, and found that they were from the `*.int.vrt` files in `interferograms` folder.

These `*.int.vrt` files return nothing when applying `GetProjection()`. I tried checking with `*.slc*` in `../merged/SLC` folder, but got nothing returned either.

It seems like these files have no projection to get, leading to this kind of error. In this case, what should I do?

Could you please provide me with some advice?

[dolphin_run_log.txt](https://github.com/user-attachments/files/17641234/dolphin_run_log.txt)
[dolphin_config.txt](https://github.com/user-attachments/files/17641334/dolphin_config.txt)

**Comment by scottstanie** [comment]
Hi, right now, the `--output-bounds` option will only work for *geocoded* inputs. Since most of the work so far has been [using Geocoded SLCs](https://www.jpl.nasa.gov/go/opera/products/cslc-product-suite/), I have not tried to add radar-to-geo conversions. 

For radar coordinate inputs, I'm assuming that a user has created whatever size SLCs they wish to run on and done their own cropping beforehand.

I do think a more helpful error early on might be useful though (as in, a check if the inputs have no geographic metadata, then error if you tried to ask for `output_bounds` or `output_bounds_wkt`.

**Comment by scottstanie** [comment]
Are the scenes very large and you wanted to run on just a subset? Or did you just provide those bounds since you wanted to see that exact area in the end?

**Reply by david-ncu2019** [reply]
Dear @scottstanie ,

First of all, really thank you for your quick response!

My purpose was to work on a subset, so I applied `--output-bounds`. Based on your comments, I guess `--output-bounds` is not for cropping the scenes.

If so, what will be a better option?

Thank you so much for your time and support!

**Reply by scottstanie** [reply]
David,
I think it's very reasonable to expect that `output_bounds` would work for radar or geo coordiantes, given how common it is to work with radar coordinates. I'll think about how I can add that with minimal additions to what dolpin knows about input formats (I'd rather avoid adding 1000s of lines of parsing code to read all possible radar processor formats). 

For now, I might suggest one of 3 things

- Mintpy has [subset.py](https://github.com/insarlab/MintPy/blob/main/src/mintpy/subset.py) for this ( though I haven't used so I can't give immediate commands for it)
- I found an old messy function from my phd to crop and isce project. It does interferograms too, but the first part is [for getting the cropped SLCs](https://github.com/scottstanie/apertools/blob/master/apertools/subset.py#L443-L472).
- If you know the row/columns within the ISCE2 outputs that you want to crop, you can do something like 
```bash
xstart=123 # starting column
ystart=456 # starting row
xsize=2000
ysize=2000

mkdir cropped-tifs
for fname in `ls merged/SLC/*/*.slc`; do
    gdal_translate $fname cropped-tifs/$(basename $fname).tif -if ISCE -projwin $xstart $ystart $xsize $ysize
done
```
and then point dolphin at `cropped-tifs/*tif`

You could also just make them `cropped-vrts` and do make the output name `.vrt` if you want to save some disk space
